### PR TITLE
oxstore command

### PR DESCRIPTION
**Pull request by rhettg**
Utility command for managing blueox logs on disk and in S3

### CODE REVIEW

File: blueox/store.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+        except ValueError:
+            pass
+
+    raise InvalidDateError()
+
+
+class LogFile(object):
+    """Represents a remote log file"""
+
+    def __init__(self, type_name, host=None, dt=None, date=None, bzip=False):
```
Comment (bickfordb) [inline blueox/store.py]: What's the difference between dt and date?

File: bin/oxstore [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+        parser.error("Missing bucket {}".format(bucket_name))
+
+    return bucket
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Manage blueox logs')
+    parser.add_argument('--verbose', '-v', action='count')
+
+    subparsers = parser.add_subparsers(
```
Comment (bickfordb) [inline bin/oxstore]: It would be nicer to use and easier to read the code if these options had help strings

**Comment by bickfordb** [comment]
r+ 

Reading through this, I was thinking that the API/code might be simpler if there was a query datastructure for reading separate from the LogFile structure.  Also, maybe it would be good to have a LocalLogFile and RemoteLogFile and then encapsulate everything inside of those?  e.g.

```python

class Query:
  def __init__(self, stream, start_at, end_at):

def open_reader(query):
   ..

def reading_example():
   for event in open_reader(Query("website_hits", Christmas, NewYears)):
        do_something(event)
  
class RemoteLogFile:
   def __init__(self, aws_conf, date):
      ..

class LocalLogFile:
    def __init__(self, date):


```

**Comment by bickfordb** [comment]
Also, as a library user it would be good to make it clear in `blueox.store` which methods are intended for export by prefixing them with underscores.

**Comment by rhettg** [comment]
@bickfordb you read my mind. I realized last night after I put this out that I still wanted a programmatic API, which was like 25% of the reason for doing this.

I'll do another round of refactoring and see if I can do that.

**Comment by bickfordb** [comment]
R+

**Pull request merged**